src/App.tsx

```diff
@@ -8,37 +8,98 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPost, getPosts } from './api/posts';
+import {
+  ErrorsNotification,
+} from './components/ErrorsNotification/ErrorsNotification';
 
 export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [error, setError] = useState('');
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loadPosts = (userId: number) => {
+    setIsLoading(true);
+
+    getPosts(userId)
+      .then(data => {
+        setPosts(data);
+      })
+      .catch(() => {
+        setError('Unable to load posts');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const loadPost = (postId: number) => {
+    getPost(postId)
+      .then(data => {
+        setSelectedPost(data);
+      })
+      .catch(() => {
+        setError('Post detail can\'t be loaded');
+      });
+  };
```

Comment:
There's no need to load individual posts from the server since you have all of them in posts state